### PR TITLE
Fix bug in `fileNameFromPosition` test helper

### PR DESCRIPTION
**Pull request by timvaillancourt**
## Description

This PR fixes a bug that causes `fileNameFromPosition` (a test helper-`func` from `go/test/endtoend/reparent/plannedreparent`) to fail to parse a filename when the position text is too long

This is causing `plannedreparent` tests to be flaky on faster, dedicated CI runners we're using because they generate more binlog positions than this test is expecting

Here is a Go Playground reproducing the problem with the old, brittle logic: https://go.dev/play/p/ck5jEMcZu_V

Thanks to @deepthi for helping me find this problem in the Vitess Slack 🙇 

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/13779

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/reparent/plannedreparent/reparent_test.go [hunk 1]
```diff
@@ -557,7 +558,16 @@ func waitForFilePosition(t *testing.T, clusterInstance *cluster.LocalProcessClus
...
 func fileNameFromPosition(pos string) string {
-	return pos[0 : len(pos)-4]
+	s := strings.SplitN(pos, ":", 2)
+	if len(s) != 2 {
+		return ""
+	}
+	return s[0]
+}
+
+func TestFileNameFromPosition(t *testing.T) {
```
Comment (timvaillancourt) [inline go/test/endtoend/reparent/plannedreparent/reparent_test.go:568]: This test is testing a test-helper, which may be overkill. Happy to remove this 👍

File: go/test/endtoend/reparent/plannedreparent/reparent_test.go [hunk 2]
```diff
@@ -557,7 +558,16 @@ func waitForFilePosition(t *testing.T, clusterInstance *cluster.LocalProcessClus
 
 // fileNameFromPosition gets the file name from the position
 func fileNameFromPosition(pos string) string {
-	return pos[0 : len(pos)-4]
```
Comment (mattlord) [inline go/test/endtoend/reparent/plannedreparent/reparent_test.go:560]: Wow, this was assuming there were 3 position digits this whole time? 🙂

**Review by mattlord — APPROVED** [review]
Thanks, @timvaillancourt !

File: go/test/endtoend/reparent/plannedreparent/reparent_test.go (hunk 2 above)
Comment (timvaillancourt) [reply go/test/endtoend/reparent/plannedreparent/reparent_test.go:560]: @mattlord yeah strange! 🙈

**Review by frouioui — APPROVED** [review]

**Pull request merged**